{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "03be69cc-02f1-42e0-afb5-ea9aa1ad99d5",
   "metadata": {},
   "source": [
    "# RNA-Seq Pipeline: Step 2 - Quality Control (QC) and Trimming\n",
    "\n",
    "This notebook runs the quality control and trimming steps on the raw `.fastq.gz` files.\n",
    "\n",
    "**Workflow:**\n",
    "1.  **Raw QC:** Run `FastQC` on the 12 compressed raw data files to assess initial quality.\n",
    "2.  **Aggregate Raw QC:** Run `MultiQC` to create a single summary report for the raw data.\n",
    "3.  **Trimming:** Use `fastp` to remove adapters, trim low-quality bases, and filter short reads.\n",
    "4.  **Trimmed QC:** Run `FastQC` again on the trimmed/cleaned files.\n",
    "5.  **Aggregate Trimmed QC:** Run `MultiQC` on the trimmed reports to assess the results of cleaning.\n",
    "\n",
    "**Tools:** `FastQC`, `MultiQC`, `fastp`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2a98d08e-6fe5-4955-be1e-449ee17f5d37",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All output directories created/verified.\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# --- Define Core Paths ---\n",
    "\n",
    "# Input directory (from previous step)\n",
    "raw_data_dir = \"01_Raw_Data\"\n",
    "\n",
    "# Output directory for all QC reports\n",
    "qc_dir = \"00_Data_QC\"\n",
    "\n",
    "# Output directory for trimmed data\n",
    "trimmed_dir = \"02_Trimmed_Data\"\n",
    "\n",
    "# --- Create Output Directories ---\n",
    "# We will create sub-directories for clarity\n",
    "\n",
    "# Directory for FastQC reports on RAW data\n",
    "raw_fastqc_dir = os.path.join(qc_dir, \"01_raw_fastqc\")\n",
    "os.makedirs(raw_fastqc_dir, exist_ok=True)\n",
    "\n",
    "# Directory for FastQC reports on TRIMMED data\n",
    "trimmed_fastqc_dir = os.path.join(qc_dir, \"02_trimmed_fastqc\")\n",
    "os.makedirs(trimmed_fastqc_dir, exist_ok=True)\n",
    "\n",
    "# Directory for the trimmed .fastq.gz files\n",
    "os.makedirs(trimmed_dir, exist_ok=True)\n",
    "\n",
    "print(f\"All output directories created/verified.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2f0f9a52-38e1-4bbd-bc40-58365d61cec8",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 1. Running FastQC on RAW Data ---\n",
      "Input Directory: 01_Raw_Data\n",
      "Output Directory: 00_Data_QC/01_raw_fastqc\n",
      "application/gzip\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of SRR34134104_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of SRR34134104_2.fastq.gz\n",
      "Started analysis of SRR34134105_1.fastq.gz\n",
      "Started analysis of SRR34134105_2.fastq.gz\n",
      "Started analysis of SRR34134106_1.fastq.gz\n",
      "Started analysis of SRR34134106_2.fastq.gz\n",
      "Started analysis of SRR34134107_1.fastq.gz\n",
      "Started analysis of SRR34134107_2.fastq.gz\n",
      "Approx 5% complete for SRR34134105_1.fastq.gz\n",
      "Approx 5% complete for SRR34134105_2.fastq.gz\n",
      "Approx 5% complete for SRR34134106_1.fastq.gz\n",
      "Approx 5% complete for SRR34134106_2.fastq.gz\n",
      "Approx 5% complete for SRR34134107_1.fastq.gz\n",
      "Approx 5% complete for SRR34134104_2.fastq.gz\n",
      "Approx 5% complete for SRR34134104_1.fastq.gz\n",
      "Approx 5% complete for SRR34134107_2.fastq.gz\n",
      "Approx 10% complete for SRR34134105_1.fastq.gz\n",
      "Approx 10% complete for SRR34134105_2.fastq.gz\n",
      "Approx 10% complete for SRR34134106_1.fastq.gz\n",
      "Approx 10% complete for SRR34134106_2.fastq.gz\n",
      "Approx 10% complete for SRR34134107_1.fastq.gz\n",
      "Approx 10% complete for SRR34134107_2.fastq.gz\n",
      "Approx 10% complete for SRR34134104_2.fastq.gz\n",
      "Approx 10% complete for SRR34134104_1.fastq.gz\n",
      "Approx 15% complete for SRR34134105_1.fastq.gz\n",
      "Approx 15% complete for SRR34134105_2.fastq.gz\n",
      "Approx 15% complete for SRR34134106_1.fastq.gz\n",
      "Approx 15% complete for SRR34134106_2.fastq.gz\n",
      "Approx 15% complete for SRR34134107_1.fastq.gz\n",
      "Approx 15% complete for SRR34134107_2.fastq.gz\n",
      "Approx 20% complete for SRR34134105_1.fastq.gz\n",
      "Approx 15% complete for SRR34134104_1.fastq.gz\n",
      "Approx 15% complete for SRR34134104_2.fastq.gz\n",
      "Approx 20% complete for SRR34134105_2.fastq.gz\n",
      "Approx 20% complete for SRR34134106_1.fastq.gz\n",
      "Approx 20% complete for SRR34134106_2.fastq.gz\n",
      "Approx 20% complete for SRR34134107_1.fastq.gz\n",
      "Approx 20% complete for SRR34134107_2.fastq.gz\n",
      "Approx 25% complete for SRR34134105_1.fastq.gz\n",
      "Approx 25% complete for SRR34134105_2.fastq.gz\n",
      "Approx 25% complete for SRR34134106_1.fastq.gz\n",
      "Approx 20% complete for SRR34134104_1.fastq.gz\n",
      "Approx 25% complete for SRR34134106_2.fastq.gz\n",
      "Approx 20% complete for SRR34134104_2.fastq.gz\n",
      "Approx 25% complete for SRR34134107_1.fastq.gz\n",
      "Approx 25% complete for SRR34134107_2.fastq.gz\n",
      "Approx 30% complete for SRR34134105_1.fastq.gz\n",
      "Approx 30% complete for SRR34134105_2.fastq.gz\n",
      "Approx 30% complete for SRR34134106_1.fastq.gz\n",
      "Approx 30% complete for SRR34134106_2.fastq.gz\n",
      "Approx 25% complete for SRR34134104_1.fastq.gz\n",
      "Approx 25% complete for SRR34134104_2.fastq.gz\n",
      "Approx 30% complete for SRR34134107_1.fastq.gz\n",
      "Approx 35% complete for SRR34134105_1.fastq.gz\n",
      "Approx 30% complete for SRR34134107_2.fastq.gz\n",
      "Approx 35% complete for SRR34134105_2.fastq.gz\n",
      "Approx 35% complete for SRR34134106_1.fastq.gz\n",
      "Approx 35% complete for SRR34134106_2.fastq.gz\n",
      "Approx 40% complete for SRR34134105_1.fastq.gz\n",
      "Approx 35% complete for SRR34134107_1.fastq.gz\n",
      "Approx 40% complete for SRR34134105_2.fastq.gz\n",
      "Approx 35% complete for SRR34134107_2.fastq.gz\n",
      "Approx 30% complete for SRR34134104_1.fastq.gz\n",
      "Approx 30% complete for SRR34134104_2.fastq.gz\n",
      "Approx 40% complete for SRR34134106_1.fastq.gz\n",
      "Approx 40% complete for SRR34134106_2.fastq.gz\n",
      "Approx 45% complete for SRR34134105_1.fastq.gz\n",
      "Approx 45% complete for SRR34134105_2.fastq.gz\n",
      "Approx 40% complete for SRR34134107_1.fastq.gz\n",
      "Approx 40% complete for SRR34134107_2.fastq.gz\n",
      "Approx 35% complete for SRR34134104_1.fastq.gz\n",
      "Approx 45% complete for SRR34134106_1.fastq.gz\n",
      "Approx 35% complete for SRR34134104_2.fastq.gz\n",
      "Approx 45% complete for SRR34134106_2.fastq.gz\n",
      "Approx 50% complete for SRR34134105_1.fastq.gz\n",
      "Approx 50% complete for SRR34134105_2.fastq.gz\n",
      "Approx 45% complete for SRR34134107_1.fastq.gz\n",
      "Approx 45% complete for SRR34134107_2.fastq.gz\n",
      "Approx 50% complete for SRR34134106_1.fastq.gz\n",
      "Approx 50% complete for SRR34134106_2.fastq.gz\n",
      "Approx 55% complete for SRR34134105_1.fastq.gz\n",
      "Approx 40% complete for SRR34134104_1.fastq.gz\n",
      "Approx 40% complete for SRR34134104_2.fastq.gz\n",
      "Approx 55% complete for SRR34134105_2.fastq.gz\n",
      "Approx 50% complete for SRR34134107_1.fastq.gz\n",
      "Approx 50% complete for SRR34134107_2.fastq.gz\n",
      "Approx 55% complete for SRR34134106_1.fastq.gz\n",
      "Approx 55% complete for SRR34134106_2.fastq.gz\n",
      "Approx 60% complete for SRR34134105_1.fastq.gz\n",
      "Approx 60% complete for SRR34134105_2.fastq.gz\n",
      "Approx 45% complete for SRR34134104_1.fastq.gz\n",
      "Approx 45% complete for SRR34134104_2.fastq.gz\n",
      "Approx 55% complete for SRR34134107_1.fastq.gz\n",
      "Approx 55% complete for SRR34134107_2.fastq.gz\n",
      "Approx 60% complete for SRR34134106_1.fastq.gz\n",
      "Approx 65% complete for SRR34134105_1.fastq.gz\n",
      "Approx 60% complete for SRR34134106_2.fastq.gz\n",
      "Approx 65% complete for SRR34134105_2.fastq.gz\n",
      "Approx 60% complete for SRR34134107_1.fastq.gz\n",
      "Approx 50% complete for SRR34134104_1.fastq.gz\n",
      "Approx 60% complete for SRR34134107_2.fastq.gz\n",
      "Approx 50% complete for SRR34134104_2.fastq.gz\n",
      "Approx 70% complete for SRR34134105_1.fastq.gz\n",
      "Approx 65% complete for SRR34134106_1.fastq.gz\n",
      "Approx 65% complete for SRR34134106_2.fastq.gz\n",
      "Approx 70% complete for SRR34134105_2.fastq.gz\n",
      "Approx 65% complete for SRR34134107_1.fastq.gz\n",
      "Approx 65% complete for SRR34134107_2.fastq.gz\n",
      "Approx 75% complete for SRR34134105_1.fastq.gz\n",
      "Approx 70% complete for SRR34134106_1.fastq.gz\n",
      "Approx 70% complete for SRR34134106_2.fastq.gz\n",
      "Approx 55% complete for SRR34134104_1.fastq.gz\n",
      "Approx 75% complete for SRR34134105_2.fastq.gz\n",
      "Approx 55% complete for SRR34134104_2.fastq.gz\n",
      "Approx 70% complete for SRR34134107_1.fastq.gz\n",
      "Approx 80% complete for SRR34134105_1.fastq.gz\n",
      "Approx 70% complete for SRR34134107_2.fastq.gz\n",
      "Approx 75% complete for SRR34134106_1.fastq.gz\n",
      "Approx 80% complete for SRR34134105_2.fastq.gz\n",
      "Approx 75% complete for SRR34134106_2.fastq.gz\n",
      "Approx 60% complete for SRR34134104_1.fastq.gz\n",
      "Approx 60% complete for SRR34134104_2.fastq.gz\n",
      "Approx 85% complete for SRR34134105_1.fastq.gz\n",
      "Approx 75% complete for SRR34134107_1.fastq.gz\n",
      "Approx 85% complete for SRR34134105_2.fastq.gz\n",
      "Approx 80% complete for SRR34134106_1.fastq.gz\n",
      "Approx 75% complete for SRR34134107_2.fastq.gz\n",
      "Approx 80% complete for SRR34134106_2.fastq.gz\n",
      "Approx 65% complete for SRR34134104_1.fastq.gz\n",
      "Approx 90% complete for SRR34134105_1.fastq.gz\n",
      "Approx 65% complete for SRR34134104_2.fastq.gz\n",
      "Approx 80% complete for SRR34134107_1.fastq.gz\n",
      "Approx 90% complete for SRR34134105_2.fastq.gz\n",
      "Approx 85% complete for SRR34134106_1.fastq.gz\n",
      "Approx 85% complete for SRR34134106_2.fastq.gz\n",
      "Approx 80% complete for SRR34134107_2.fastq.gz\n",
      "Approx 95% complete for SRR34134105_1.fastq.gz\n",
      "Approx 95% complete for SRR34134105_2.fastq.gz\n",
      "Approx 70% complete for SRR34134104_1.fastq.gz\n",
      "Approx 85% complete for SRR34134107_1.fastq.gz\n",
      "Approx 90% complete for SRR34134106_1.fastq.gz\n",
      "Approx 90% complete for SRR34134106_2.fastq.gz\n",
      "Approx 85% complete for SRR34134107_2.fastq.gz\n",
      "Approx 70% complete for SRR34134104_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134105_1.fastq.gz\n",
      "Analysis complete for SRR34134105_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Approx 95% complete for SRR34134106_1.fastq.gz\n",
      "Approx 95% complete for SRR34134106_2.fastq.gz\n",
      "Approx 90% complete for SRR34134107_1.fastq.gz\n",
      "Started analysis of SRR34134108_1.fastq.gz\n",
      "Started analysis of SRR34134108_2.fastq.gz\n",
      "Approx 90% complete for SRR34134107_2.fastq.gz\n",
      "Approx 75% complete for SRR34134104_1.fastq.gz\n",
      "Approx 75% complete for SRR34134104_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134106_1.fastq.gz\n",
      "Analysis complete for SRR34134106_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Approx 5% complete for SRR34134108_1.fastq.gz\n",
      "Approx 95% complete for SRR34134107_1.fastq.gz\n",
      "Approx 5% complete for SRR34134108_2.fastq.gz\n",
      "Started analysis of SRR34134109_1.fastq.gz\n",
      "Approx 95% complete for SRR34134107_2.fastq.gz\n",
      "Started analysis of SRR34134109_2.fastq.gz\n",
      "Approx 80% complete for SRR34134104_1.fastq.gz\n",
      "Approx 80% complete for SRR34134104_2.fastq.gz\n",
      "Approx 10% complete for SRR34134108_1.fastq.gz\n",
      "Approx 10% complete for SRR34134108_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134107_1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Approx 5% complete for SRR34134109_1.fastq.gz\n",
      "Approx 5% complete for SRR34134109_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134107_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Approx 85% complete for SRR34134104_1.fastq.gz\n",
      "Approx 85% complete for SRR34134104_2.fastq.gz\n",
      "Approx 15% complete for SRR34134108_2.fastq.gz\n",
      "Approx 15% complete for SRR34134108_1.fastq.gz\n",
      "Approx 10% complete for SRR34134109_2.fastq.gz\n",
      "Approx 10% complete for SRR34134109_1.fastq.gz\n",
      "Approx 90% complete for SRR34134104_1.fastq.gz\n",
      "Approx 90% complete for SRR34134104_2.fastq.gz\n",
      "Approx 20% complete for SRR34134108_1.fastq.gz\n",
      "Approx 20% complete for SRR34134108_2.fastq.gz\n",
      "Approx 15% complete for SRR34134109_2.fastq.gz\n",
      "Approx 15% complete for SRR34134109_1.fastq.gz\n",
      "Approx 25% complete for SRR34134108_1.fastq.gz\n",
      "Approx 95% complete for SRR34134104_1.fastq.gz\n",
      "Approx 20% complete for SRR34134109_1.fastq.gz\n",
      "Approx 95% complete for SRR34134104_2.fastq.gz\n",
      "Approx 25% complete for SRR34134108_2.fastq.gz\n",
      "Approx 20% complete for SRR34134109_2.fastq.gz\n",
      "Approx 30% complete for SRR34134108_1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134104_1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Approx 25% complete for SRR34134109_1.fastq.gz\n",
      "Approx 25% complete for SRR34134109_2.fastq.gz\n",
      "Approx 30% complete for SRR34134108_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134104_2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Approx 35% complete for SRR34134108_1.fastq.gz\n",
      "Approx 30% complete for SRR34134109_1.fastq.gz\n",
      "Approx 30% complete for SRR34134109_2.fastq.gz\n",
      "Approx 35% complete for SRR34134108_2.fastq.gz\n",
      "Approx 40% complete for SRR34134108_1.fastq.gz\n",
      "Approx 35% complete for SRR34134109_1.fastq.gz\n",
      "Approx 35% complete for SRR34134109_2.fastq.gz\n",
      "Approx 40% complete for SRR34134108_2.fastq.gz\n",
      "Approx 45% complete for SRR34134108_1.fastq.gz\n",
      "Approx 40% complete for SRR34134109_1.fastq.gz\n",
      "Approx 40% complete for SRR34134109_2.fastq.gz\n",
      "Approx 45% complete for SRR34134108_2.fastq.gz\n",
      "Approx 50% complete for SRR34134108_1.fastq.gz\n",
      "Approx 45% complete for SRR34134109_1.fastq.gz\n",
      "Approx 45% complete for SRR34134109_2.fastq.gz\n",
      "Approx 50% complete for SRR34134108_2.fastq.gz\n",
      "Approx 55% complete for SRR34134108_1.fastq.gz\n",
      "Approx 50% complete for SRR34134109_1.fastq.gz\n",
      "Approx 50% complete for SRR34134109_2.fastq.gz\n",
      "Approx 55% complete for SRR34134108_2.fastq.gz\n",
      "Approx 60% complete for SRR34134108_1.fastq.gz\n",
      "Approx 55% complete for SRR34134109_1.fastq.gz\n",
      "Approx 55% complete for SRR34134109_2.fastq.gz\n",
      "Approx 60% complete for SRR34134108_2.fastq.gz\n",
      "Approx 65% complete for SRR34134108_1.fastq.gz\n",
      "Approx 60% complete for SRR34134109_1.fastq.gz\n",
      "Approx 60% complete for SRR34134109_2.fastq.gz\n",
      "Approx 65% complete for SRR34134108_2.fastq.gz\n",
      "Approx 70% complete for SRR34134108_1.fastq.gz\n",
      "Approx 65% complete for SRR34134109_1.fastq.gz\n",
      "Approx 65% complete for SRR34134109_2.fastq.gz\n",
      "Approx 70% complete for SRR34134108_2.fastq.gz\n",
      "Approx 75% complete for SRR34134108_1.fastq.gz\n",
      "Approx 70% complete for SRR34134109_1.fastq.gz\n",
      "Approx 70% complete for SRR34134109_2.fastq.gz\n",
      "Approx 75% complete for SRR34134108_2.fastq.gz\n",
      "Approx 80% complete for SRR34134108_1.fastq.gz\n",
      "Approx 75% complete for SRR34134109_1.fastq.gz\n",
      "Approx 75% complete for SRR34134109_2.fastq.gz\n",
      "Approx 80% complete for SRR34134108_2.fastq.gz\n",
      "Approx 85% complete for SRR34134108_1.fastq.gz\n",
      "Approx 80% complete for SRR34134109_1.fastq.gz\n",
      "Approx 80% complete for SRR34134109_2.fastq.gz\n",
      "Approx 85% complete for SRR34134108_2.fastq.gz\n",
      "Approx 90% complete for SRR34134108_1.fastq.gz\n",
      "Approx 85% complete for SRR34134109_1.fastq.gz\n",
      "Approx 85% complete for SRR34134109_2.fastq.gz\n",
      "Approx 90% complete for SRR34134108_2.fastq.gz\n",
      "Approx 95% complete for SRR34134108_1.fastq.gz\n",
      "Approx 90% complete for SRR34134109_1.fastq.gz\n",
      "Approx 90% complete for SRR34134109_2.fastq.gz\n",
      "Approx 95% complete for SRR34134108_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134108_1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Approx 95% complete for SRR34134109_1.fastq.gz\n",
      "Approx 95% complete for SRR34134109_2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134108_2.fastq.gz\n",
      "Analysis complete for SRR34134109_1.fastq.gz\n",
      "Analysis complete for SRR34134109_2.fastq.gz\n",
      "--- Raw FastQC complete. ---\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$raw_data_dir\" \"$raw_fastqc_dir\"\n",
    "# $1 = raw_data_dir\n",
    "# $2 = raw_fastqc_dir\n",
    "\n",
    "RAW_DIR=$1\n",
    "RAW_QC_OUT=$2\n",
    "\n",
    "echo \"--- 1. Running FastQC on RAW Data ---\"\n",
    "echo \"Input Directory: $RAW_DIR\"\n",
    "echo \"Output Directory: $RAW_QC_OUT\"\n",
    "\n",
    "# Run FastQC\n",
    "# -t 8 : Use 8 threads\n",
    "# -o $RAW_QC_OUT : Output directory\n",
    "# $RAW_DIR/*.fastq.gz : Run on all .fastq.gz files in the input directory\n",
    "fastqc -t 8 -o $RAW_QC_OUT $RAW_DIR/*.fastq.gz\n",
    "\n",
    "echo \"--- Raw FastQC complete. ---\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0f532056-5a85-45a3-b4bc-b7b50aaaa904",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 2. Running MultiQC on RAW FastQC Reports ---\n",
      "Target Directory: 00_Data_QC/01_raw_fastqc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\u001b[91m///\u001b[0m \u001b]8;id=934496;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ ðŸ” \u001b[2mv1.30\u001b[0m\n",
      "\n",
      "\u001b[34m     version_check\u001b[0m | \u001b[33mMultiQC Version v1.31 now available!\u001b[0m\n",
      "\u001b[34m       file_search\u001b[0m | Search path: /home/refm_youssef/rnaseq_project/00_Data_QC/01_raw_fastqc\n",
      "\u001b[2K         \u001b[34msearching\u001b[0m | \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m28/28\u001b[0m  \u001b[32m23/28\u001b[0m \u001b[2m00_Data_QC/01_raw_fastqc/SRR34134108_1_fastqc.html\u001b[0m\n",
      "\u001b[?25h\u001b[34m            fastqc\u001b[0m | Found 12 reports\n",
      "\u001b[34m     write_results\u001b[0m | Existing reports found, adding suffix to filenames. Use '--force' to overwrite.\n",
      "\u001b[34m     write_results\u001b[0m | Data        : 00_Data_QC/01_raw_fastqc/multiqc_data_1\n",
      "\u001b[34m     write_results\u001b[0m | Report      : 00_Data_QC/01_raw_fastqc/multiqc_report_1.html\n",
      "\u001b[34m           multiqc\u001b[0m | MultiQC complete\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- Raw MultiQC complete. ---\n",
      "Check the 'multiqc_report.html' file in 00_Data_QC/01_raw_fastqc\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$raw_fastqc_dir\"\n",
    "# $1 = raw_fastqc_dir (this variable was set in Cell 2)\n",
    "\n",
    "RAW_QC_OUT=$1\n",
    "\n",
    "echo \"--- 2. Running MultiQC on RAW FastQC Reports ---\"\n",
    "echo \"Target Directory: $RAW_QC_OUT\"\n",
    "\n",
    "# Run MultiQC\n",
    "# -o $RAW_QC_OUT : Output directory\n",
    "# $RAW_QC_OUT : Directory to scan for reports\n",
    "multiqc -o $RAW_QC_OUT $RAW_QC_OUT\n",
    "\n",
    "echo \"--- Raw MultiQC complete. ---\"\n",
    "echo \"Check the 'multiqc_report.html' file in $RAW_QC_OUT\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5093fb3f-b643-4d56-87cb-792b0d1d9033",
   "metadata": {},
   "source": [
    "### 3. Trimming and Filtering (fastp)\n",
    "\n",
    "**Analysis of Raw QC Report (multiqc_report.html):**\n",
    "* **Good News:** The overall sequence quality (`Per Base Sequence Quality`) is excellent.\n",
    "* **Problem 1 (High Priority):** The `Adapter Content` plot shows significant **Illumina adapter contamination**. These must be removed.\n",
    "* **Problem 2 (Medium Priority):** The `Per Base Sequence Content` plot shows a strong bias in the first ~10-15 bases (a common **random primer** artifact).\n",
    "\n",
    "**Action:**\n",
    "We will use `fastp` to clean the data by:\n",
    "1.  Removing adapters automatically.\n",
    "2.  Trimming the first 10 bases from both Read 1 and Read 2 (`--trim_front1=10`, `--trim_front2=10`) to remove the bias.\n",
    "3.  Trimming low-quality bases from the ends."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ce7769ee-7e5e-4c01-a152-75655ea7041e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 6 samples to trim: ['SRR34134104', 'SRR34134105', 'SRR34134106', 'SRR34134107', 'SRR34134108', 'SRR34134109']\n",
      "\n",
      "--- Starting fastp Trimming Loop ---\n",
      "Processing sample: SRR34134104 ...\n",
      "Detecting adapter sequence for read1...\n",
      ">Nextera_LMP_Read1_External_Adapter | >Illumina Multiplexing Index Sequencing Primer\n",
      "GATCGGAAGAGCACACGTCTGAACTCCAGTCAC\n",
      "\n",
      "Detecting adapter sequence for read2...\n",
      ">Illumina TruSeq Adapter Read 2\n",
      "AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 20798968\n",
      "total bases: 3119845200\n",
      "Q20 bases: 3067097368(98.3093%)\n",
      "Q30 bases: 2962826594(94.9671%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 20798968\n",
      "total bases: 3119845200\n",
      "Q20 bases: 3046404045(97.646%)\n",
      "Q30 bases: 2915427732(93.4478%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 20728434\n",
      "total bases: 2847514975\n",
      "Q20 bases: 2802226697(98.4096%)\n",
      "Q30 bases: 2708023546(95.1013%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 20728434\n",
      "total bases: 2847468201\n",
      "Q20 bases: 2792856725(98.0821%)\n",
      "Q30 bases: 2675600566(93.9642%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 41456868\n",
      "reads failed due to low quality: 138416\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 2652\n",
      "reads with adapter trimmed: 3555177\n",
      "bases trimmed due to adapters: 109544206\n",
      "\n",
      "Duplication rate: 58.8983%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 269\n",
      "\n",
      "JSON report: 02_Trimmed_Data/SRR34134104.fastp.json\n",
      "HTML report: 02_Trimmed_Data/SRR34134104.fastp.html\n",
      "\n",
      "fastp --in1 01_Raw_Data/SRR34134104_1.fastq.gz --in2 01_Raw_Data/SRR34134104_2.fastq.gz --out1 02_Trimmed_Data/SRR34134104.trimmed_R1.fastq.gz --out2 02_Trimmed_Data/SRR34134104.trimmed_R2.fastq.gz --html 02_Trimmed_Data/SRR34134104.fastp.html --json 02_Trimmed_Data/SRR34134104.fastp.json --thread 8 --detect_adapter_for_pe --trim_front1 10 --trim_front2 10 \n",
      "fastp v1.0.1, time used: 174 seconds\n",
      "Finished trimming SRR34134104.\n",
      "Processing sample: SRR34134105 ...\n",
      "Detecting adapter sequence for read1...\n",
      ">Illumina TruSeq Adapter Read 1\n",
      "AGATCGGAAGAGCACACGTCTGAACTCCAGTCA\n",
      "\n",
      "Detecting adapter sequence for read2...\n",
      ">Illumina TruSeq Adapter Read 2\n",
      "AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 15131966\n",
      "total bases: 2269794900\n",
      "Q20 bases: 2231031874(98.2922%)\n",
      "Q30 bases: 2155300679(94.9557%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 15131966\n",
      "total bases: 2269794900\n",
      "Q20 bases: 2216488758(97.6515%)\n",
      "Q30 bases: 2123615468(93.5598%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 15080238\n",
      "total bases: 2065648515\n",
      "Q20 bases: 2033212972(98.4298%)\n",
      "Q30 bases: 1965451795(95.1494%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 15080238\n",
      "total bases: 2065691055\n",
      "Q20 bases: 2027338754(98.1434%)\n",
      "Q30 bases: 1944967960(94.1558%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 30160476\n",
      "reads failed due to low quality: 101532\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1924\n",
      "reads with adapter trimmed: 2894974\n",
      "bases trimmed due to adapters: 91549366\n",
      "\n",
      "Duplication rate: 55.4631%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 269\n",
      "\n",
      "JSON report: 02_Trimmed_Data/SRR34134105.fastp.json\n",
      "HTML report: 02_Trimmed_Data/SRR34134105.fastp.html\n",
      "\n",
      "fastp --in1 01_Raw_Data/SRR34134105_1.fastq.gz --in2 01_Raw_Data/SRR34134105_2.fastq.gz --out1 02_Trimmed_Data/SRR34134105.trimmed_R1.fastq.gz --out2 02_Trimmed_Data/SRR34134105.trimmed_R2.fastq.gz --html 02_Trimmed_Data/SRR34134105.fastp.html --json 02_Trimmed_Data/SRR34134105.fastp.json --thread 8 --detect_adapter_for_pe --trim_front1 10 --trim_front2 10 \n",
      "fastp v1.0.1, time used: 127 seconds\n",
      "Finished trimming SRR34134105.\n",
      "Processing sample: SRR34134106 ...\n",
      "Detecting adapter sequence for read1...\n",
      ">Illumina TruSeq Adapter Read 1\n",
      "AGATCGGAAGAGCACACGTCTGAACTCCAGTCA\n",
      "\n",
      "Detecting adapter sequence for read2...\n",
      ">Illumina TruSeq Adapter Read 2\n",
      "AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 16185133\n",
      "total bases: 2427769950\n",
      "Q20 bases: 2400045309(98.858%)\n",
      "Q30 bases: 2339544883(96.366%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 16185133\n",
      "total bases: 2427769950\n",
      "Q20 bases: 2390089081(98.4479%)\n",
      "Q30 bases: 2314292501(95.3259%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 16117073\n",
      "total bases: 2246452735\n",
      "Q20 bases: 2221857224(98.9051%)\n",
      "Q30 bases: 2165912241(96.4148%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 16117073\n",
      "total bases: 2246601133\n",
      "Q20 bases: 2217003616(98.6826%)\n",
      "Q30 bases: 2147442376(95.5863%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 32234146\n",
      "reads failed due to low quality: 129418\n",
      "reads failed due to too many N: 6500\n",
      "reads failed due to too short: 202\n",
      "reads with adapter trimmed: 1104116\n",
      "bases trimmed due to adapters: 19813089\n",
      "\n",
      "Duplication rate: 53.3266%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 269\n",
      "\n",
      "JSON report: 02_Trimmed_Data/SRR34134106.fastp.json\n",
      "HTML report: 02_Trimmed_Data/SRR34134106.fastp.html\n",
      "\n",
      "fastp --in1 01_Raw_Data/SRR34134106_1.fastq.gz --in2 01_Raw_Data/SRR34134106_2.fastq.gz --out1 02_Trimmed_Data/SRR34134106.trimmed_R1.fastq.gz --out2 02_Trimmed_Data/SRR34134106.trimmed_R2.fastq.gz --html 02_Trimmed_Data/SRR34134106.fastp.html --json 02_Trimmed_Data/SRR34134106.fastp.json --thread 8 --detect_adapter_for_pe --trim_front1 10 --trim_front2 10 \n",
      "fastp v1.0.1, time used: 144 seconds\n",
      "Finished trimming SRR34134106.\n",
      "Processing sample: SRR34134107 ...\n",
      "Detecting adapter sequence for read1...\n",
      ">Nextera_LMP_Read1_External_Adapter | >Illumina Multiplexing Index Sequencing Primer\n",
      "GATCGGAAGAGCACACGTCTGAACTCCAGTCAC\n",
      "\n",
      "Detecting adapter sequence for read2...\n",
      ">Illumina TruSeq Adapter Read 2\n",
      "AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 17243686\n",
      "total bases: 2586552900\n",
      "Q20 bases: 2556574941(98.841%)\n",
      "Q30 bases: 2492784555(96.3748%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 17243686\n",
      "total bases: 2586552900\n",
      "Q20 bases: 2550852258(98.6198%)\n",
      "Q30 bases: 2479961185(95.879%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 17183492\n",
      "total bases: 2382290034\n",
      "Q20 bases: 2355878403(98.8913%)\n",
      "Q30 bases: 2297234155(96.4297%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 17183492\n",
      "total bases: 2382277304\n",
      "Q20 bases: 2356317740(98.9103%)\n",
      "Q30 bases: 2292309100(96.2234%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 34366984\n",
      "reads failed due to low quality: 113008\n",
      "reads failed due to too many N: 7166\n",
      "reads failed due to too short: 214\n",
      "reads with adapter trimmed: 2377070\n",
      "bases trimmed due to adapters: 46931447\n",
      "\n",
      "Duplication rate: 66.0083%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 269\n",
      "\n",
      "JSON report: 02_Trimmed_Data/SRR34134107.fastp.json\n",
      "HTML report: 02_Trimmed_Data/SRR34134107.fastp.html\n",
      "\n",
      "fastp --in1 01_Raw_Data/SRR34134107_1.fastq.gz --in2 01_Raw_Data/SRR34134107_2.fastq.gz --out1 02_Trimmed_Data/SRR34134107.trimmed_R1.fastq.gz --out2 02_Trimmed_Data/SRR34134107.trimmed_R2.fastq.gz --html 02_Trimmed_Data/SRR34134107.fastp.html --json 02_Trimmed_Data/SRR34134107.fastp.json --thread 8 --detect_adapter_for_pe --trim_front1 10 --trim_front2 10 \n",
      "fastp v1.0.1, time used: 146 seconds\n",
      "Finished trimming SRR34134107.\n",
      "Processing sample: SRR34134108 ...\n",
      "Detecting adapter sequence for read1...\n",
      ">Illumina TruSeq Adapter Read 1\n",
      "AGATCGGAAGAGCACACGTCTGAACTCCAGTCA\n",
      "\n",
      "Detecting adapter sequence for read2...\n",
      ">Illumina TruSeq Adapter Read 2\n",
      "AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 17765045\n",
      "total bases: 2664756750\n",
      "Q20 bases: 2616695127(98.1964%)\n",
      "Q30 bases: 2519862218(94.5626%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 17765045\n",
      "total bases: 2664756750\n",
      "Q20 bases: 2606055466(97.7971%)\n",
      "Q30 bases: 2496381451(93.6814%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 17703292\n",
      "total bases: 2456133538\n",
      "Q20 bases: 2413064340(98.2465%)\n",
      "Q30 bases: 2323765355(94.6107%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 17703292\n",
      "total bases: 2456070095\n",
      "Q20 bases: 2408230537(98.0522%)\n",
      "Q30 bases: 2307735373(93.9605%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 35406584\n",
      "reads failed due to low quality: 123300\n",
      "reads failed due to too many N: 12\n",
      "reads failed due to too short: 194\n",
      "reads with adapter trimmed: 2232974\n",
      "bases trimmed due to adapters: 44843468\n",
      "\n",
      "Duplication rate: 62.3397%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 269\n",
      "\n",
      "JSON report: 02_Trimmed_Data/SRR34134108.fastp.json\n",
      "HTML report: 02_Trimmed_Data/SRR34134108.fastp.html\n",
      "\n",
      "fastp --in1 01_Raw_Data/SRR34134108_1.fastq.gz --in2 01_Raw_Data/SRR34134108_2.fastq.gz --out1 02_Trimmed_Data/SRR34134108.trimmed_R1.fastq.gz --out2 02_Trimmed_Data/SRR34134108.trimmed_R2.fastq.gz --html 02_Trimmed_Data/SRR34134108.fastp.html --json 02_Trimmed_Data/SRR34134108.fastp.json --thread 8 --detect_adapter_for_pe --trim_front1 10 --trim_front2 10 \n",
      "fastp v1.0.1, time used: 153 seconds\n",
      "Finished trimming SRR34134108.\n",
      "Processing sample: SRR34134109 ...\n",
      "Detecting adapter sequence for read1...\n",
      ">Nextera_LMP_Read1_External_Adapter | >Illumina Multiplexing Index Sequencing Primer\n",
      "GATCGGAAGAGCACACGTCTGAACTCCAGTCAC\n",
      "\n",
      "Detecting adapter sequence for read2...\n",
      ">Illumina TruSeq Adapter Read 2\n",
      "AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 17520984\n",
      "total bases: 2628147600\n",
      "Q20 bases: 2582067514(98.2467%)\n",
      "Q30 bases: 2488498126(94.6864%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 17520984\n",
      "total bases: 2628147600\n",
      "Q20 bases: 2557262755(97.3029%)\n",
      "Q30 bases: 2430655485(92.4855%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 17441876\n",
      "total bases: 2432438917\n",
      "Q20 bases: 2391139885(98.3022%)\n",
      "Q30 bases: 2304617988(94.7452%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 17441876\n",
      "total bases: 2432384096\n",
      "Q20 bases: 2371953023(97.5156%)\n",
      "Q30 bases: 2254929414(92.7045%)\n",
      "Q40 bases: 0(0%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 34883752\n",
      "reads failed due to low quality: 158134\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 82\n",
      "reads with adapter trimmed: 1089702\n",
      "bases trimmed due to adapters: 18967697\n",
      "\n",
      "Duplication rate: 60.1392%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 269\n",
      "\n",
      "JSON report: 02_Trimmed_Data/SRR34134109.fastp.json\n",
      "HTML report: 02_Trimmed_Data/SRR34134109.fastp.html\n",
      "\n",
      "fastp --in1 01_Raw_Data/SRR34134109_1.fastq.gz --in2 01_Raw_Data/SRR34134109_2.fastq.gz --out1 02_Trimmed_Data/SRR34134109.trimmed_R1.fastq.gz --out2 02_Trimmed_Data/SRR34134109.trimmed_R2.fastq.gz --html 02_Trimmed_Data/SRR34134109.fastp.html --json 02_Trimmed_Data/SRR34134109.fastp.json --thread 8 --detect_adapter_for_pe --trim_front1 10 --trim_front2 10 \n",
      "fastp v1.0.1, time used: 157 seconds\n",
      "Finished trimming SRR34134109.\n",
      "--- fastp Trimming complete. ---\n",
      "Trimmed files are in: 02_Trimmed_Data\n"
     ]
    }
   ],
   "source": [
    "# We will use Python to create the loop.\n",
    "\n",
    "import os\n",
    "\n",
    "# --- Get Sample Names ---\n",
    "# We get the variables (like raw_data_dir) from Cell 2\n",
    "# We only need the R1 files to get the sample names\n",
    "input_files = sorted([f for f in os.listdir(raw_data_dir) if f.endswith(\"_1.fastq.gz\")])\n",
    "\n",
    "# Get just the sample names (e.g., \"SRR34134109\")\n",
    "sample_names = [f.split(\"_1.fastq.gz\")[0] for f in input_files]\n",
    "\n",
    "print(f\"Found {len(sample_names)} samples to trim: {sample_names}\")\n",
    "\n",
    "# --- Start the loop ---\n",
    "print(\"\\n--- Starting fastp Trimming Loop ---\")\n",
    "\n",
    "for sample in sample_names:\n",
    "    print(f\"Processing sample: {sample} ...\")\n",
    "    \n",
    "    # Define input paths for R1 and R2\n",
    "    in_r1 = f\"{raw_data_dir}/{sample}_1.fastq.gz\"\n",
    "    in_r2 = f\"{raw_data_dir}/{sample}_2.fastq.gz\"\n",
    "    \n",
    "    # Define output paths for cleaned R1 and R2\n",
    "    out_r1 = f\"{trimmed_dir}/{sample}.trimmed_R1.fastq.gz\"\n",
    "    out_r2 = f\"{trimmed_dir}/{sample}.trimmed_R2.fastq.gz\"\n",
    "    \n",
    "    # Define paths for the reports that fastp creates\n",
    "    report_html = f\"{trimmed_dir}/{sample}.fastp.html\"\n",
    "    report_json = f\"{trimmed_dir}/{sample}.fastp.json\"\n",
    "    \n",
    "    # Build the fastp command\n",
    "    # We use '!' to run the command in the shell\n",
    "    !fastp \\\n",
    "        --in1 $in_r1 \\\n",
    "        --in2 $in_r2 \\\n",
    "        --out1 $out_r1 \\\n",
    "        --out2 $out_r2 \\\n",
    "        --html $report_html \\\n",
    "        --json $report_json \\\n",
    "        --thread 8 \\\n",
    "        --detect_adapter_for_pe \\\n",
    "        --trim_front1 10 \\\n",
    "        --trim_front2 10\n",
    "        \n",
    "    print(f\"Finished trimming {sample}.\")\n",
    "\n",
    "print(\"--- fastp Trimming complete. ---\")\n",
    "print(f\"Trimmed files are in: {trimmed_dir}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "24000119-dbc2-4005-9fc6-7f7f93b4acbd",
   "metadata": {},
   "source": [
    "### 4. Post-Trimming Quality Control\n",
    "\n",
    "Now that `fastp` has finished, we have a new set of cleaned `.fastq.gz` files in the `02_Trimmed_Data` directory.\n",
    "\n",
    "We must run `FastQC` and `MultiQC` **again** on this new data to:\n",
    "1.  **Verify** that the adapters are gone.\n",
    "2.  **Confirm** that the random primer bias (first 10 bases) is gone.\n",
    "3.  **Ensure** that we didn't introduce any new problems."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e1981bd2-9281-4b15-933b-3a1ddc380efd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 4. Running FastQC on TRIMMED Data ---\n",
      "Input Directory: 02_Trimmed_Data\n",
      "Output Directory: 00_Data_QC/02_trimmed_fastqc\n",
      "application/gzip\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of SRR34134104.trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of SRR34134104.trimmed_R2.fastq.gz\n",
      "Started analysis of SRR34134105.trimmed_R1.fastq.gz\n",
      "Started analysis of SRR34134105.trimmed_R2.fastq.gz\n",
      "Started analysis of SRR34134106.trimmed_R1.fastq.gz\n",
      "Started analysis of SRR34134106.trimmed_R2.fastq.gz\n",
      "Started analysis of SRR34134107.trimmed_R1.fastq.gz\n",
      "Started analysis of SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 5% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 5% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 5% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 5% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Approx 5% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 5% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 5% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 5% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 10% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 10% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 10% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 10% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Approx 10% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 10% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 10% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 10% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 15% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 15% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 15% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 15% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Approx 15% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 15% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 15% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 20% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 20% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 15% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 20% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 20% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Approx 20% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 20% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 25% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 25% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 20% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 25% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 20% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 25% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Approx 25% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 25% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 30% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 30% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 30% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 30% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Approx 25% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 25% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 30% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 35% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 30% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 35% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 35% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 35% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Approx 40% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 35% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 40% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 30% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 35% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 30% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 40% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 40% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Approx 45% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 45% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 40% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 40% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 35% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 35% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 45% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 45% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Approx 50% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 50% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 45% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 45% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 40% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 50% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 50% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Approx 40% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 55% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 55% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 50% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 50% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 55% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 55% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Approx 60% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 60% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 45% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 45% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 55% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 55% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 60% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 65% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 60% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Approx 65% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 50% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 60% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 50% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 60% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 65% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 70% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 70% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 65% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Approx 65% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 65% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 55% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 75% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 55% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 70% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 75% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 70% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Approx 70% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 80% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 70% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 80% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 75% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 75% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Approx 60% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 60% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 85% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 75% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 85% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 75% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 80% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 80% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Approx 65% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 65% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 90% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 90% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 80% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 80% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 85% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 85% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Approx 95% complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Approx 95% complete for SRR34134105.trimmed_R2.fastq.gz\n",
      "Approx 70% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 70% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 85% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 90% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 85% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 90% complete for SRR34134106.trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134105.trimmed_R1.fastq.gz\n",
      "Analysis complete for SRR34134105.trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Approx 95% complete for SRR34134106.trimmed_R1.fastq.gz\n",
      "Approx 75% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 90% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Started analysis of SRR34134108.trimmed_R1.fastq.gz\n",
      "Approx 90% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 75% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 95% complete for SRR34134106.trimmed_R2.fastq.gz\n",
      "Started analysis of SRR34134108.trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134106.trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Approx 5% complete for SRR34134108.trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134106.trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Approx 95% complete for SRR34134107.trimmed_R2.fastq.gz\n",
      "Approx 95% complete for SRR34134107.trimmed_R1.fastq.gz\n",
      "Approx 5% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Started analysis of SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 80% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 80% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Started analysis of SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 10% complete for SRR34134108.trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134107.trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Approx 10% complete for SRR34134108.trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134107.trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Approx 5% complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 5% complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 85% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 85% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 15% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Approx 15% complete for SRR34134108.trimmed_R1.fastq.gz\n",
      "Approx 10% complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 10% complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 90% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 90% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 20% complete for SRR34134108.trimmed_R1.fastq.gz\n",
      "Approx 15% complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 20% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Approx 15% complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 95% complete for SRR34134104.trimmed_R2.fastq.gz\n",
      "Approx 95% complete for SRR34134104.trimmed_R1.fastq.gz\n",
      "Approx 25% complete for SRR34134108.trimmed_R1.fastq.gz\n",
      "Approx 20% complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 25% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Approx 20% complete for SRR34134109.trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134104.trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Approx 30% complete for SRR34134108.trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134104.trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Approx 30% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Approx 25% complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 25% complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 35% complete for SRR34134108.trimmed_R1.fastq.gz\n",
      "Approx 35% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Approx 30% complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 30% complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 40% complete for SRR34134108.trimmed_R1.fastq.gz\n",
      "Approx 40% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Approx 35% complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 35% complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 45% complete for SRR34134108.trimmed_R1.fastq.gz\n",
      "Approx 45% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Approx 40% complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 40% complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 50% complete for SRR34134108.trimmed_R1.fastq.gz\n",
      "Approx 45% complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 50% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Approx 45% complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 55% complete for SRR34134108.trimmed_R1.fastq.gz\n",
      "Approx 50% complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 55% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Approx 50% complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 60% complete for SRR34134108.trimmed_R1.fastq.gz\n",
      "Approx 55% complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 60% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Approx 55% complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 65% complete for SRR34134108.trimmed_R1.fastq.gz\n",
      "Approx 60% complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 65% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Approx 60% complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 70% complete for SRR34134108.trimmed_R1.fastq.gz\n",
      "Approx 65% complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 70% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Approx 65% complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 75% complete for SRR34134108.trimmed_R1.fastq.gz\n",
      "Approx 70% complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 75% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Approx 70% complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 80% complete for SRR34134108.trimmed_R1.fastq.gz\n",
      "Approx 75% complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 80% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Approx 75% complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 85% complete for SRR34134108.trimmed_R1.fastq.gz\n",
      "Approx 80% complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 85% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Approx 80% complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 90% complete for SRR34134108.trimmed_R1.fastq.gz\n",
      "Approx 85% complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 90% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Approx 85% complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "Approx 95% complete for SRR34134108.trimmed_R1.fastq.gz\n",
      "Approx 90% complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Approx 95% complete for SRR34134108.trimmed_R2.fastq.gz\n",
      "Approx 90% complete for SRR34134109.trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134108.trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Approx 95% complete for SRR34134109.trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134108.trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Approx 95% complete for SRR34134109.trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR34134109.trimmed_R1.fastq.gz\n",
      "Analysis complete for SRR34134109.trimmed_R2.fastq.gz\n",
      "--- Trimmed FastQC complete. ---\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$trimmed_dir\" \"$trimmed_fastqc_dir\"\n",
    "# $1 = trimmed_dir (variable from Cell 2)\n",
    "# $2 = trimmed_fastqc_dir (variable from Cell 2)\n",
    "\n",
    "TRIMMED_DIR=$1\n",
    "TRIMMED_QC_OUT=$2\n",
    "\n",
    "echo \"--- 4. Running FastQC on TRIMMED Data ---\"\n",
    "echo \"Input Directory: $TRIMMED_DIR\"\n",
    "echo \"Output Directory: $TRIMMED_QC_OUT\"\n",
    "\n",
    "# Run FastQC\n",
    "# -t 8 : Use 8 threads\n",
    "# -o $TRIMMED_QC_OUT : Output directory\n",
    "# $TRIMMED_DIR/*.fastq.gz : Run on all .fastq.gz files in the TRIMMED directory\n",
    "fastqc -t 8 -o $TRIMMED_QC_OUT $TRIMMED_DIR/*.fastq.gz\n",
    "\n",
    "echo \"--- Trimmed FastQC complete. ---\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3738dbd0-76f7-434f-b6a3-04940b52a822",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 5. Running MultiQC on TRIMMED FastQC Reports ---\n",
      "Target Directory: 00_Data_QC/02_trimmed_fastqc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\u001b[91m///\u001b[0m \u001b]8;id=735404;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ ðŸ” \u001b[2mv1.30\u001b[0m\n",
      "\n",
      "\u001b[34m     version_check\u001b[0m | \u001b[33mMultiQC Version v1.31 now available!\u001b[0m\n",
      "\u001b[34m       file_search\u001b[0m | Search path: /home/refm_youssef/rnaseq_project/00_Data_QC/02_trimmed_fastqc\n",
      "\u001b[2K         \u001b[34msearching\u001b[0m | \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m24/24\u001b[0m  \u001b[32m21/24\u001b[0m \u001b[2m_trimmed_fastqc/SRR34134109.trimmed_R2_fastqc.html\u001b[0m\n",
      "\u001b[?25h\u001b[34m            fastqc\u001b[0m | Found 6 reports\n",
      "\u001b[34m     write_results\u001b[0m | Data        : 00_Data_QC/02_trimmed_fastqc/multiqc_data\n",
      "\u001b[34m     write_results\u001b[0m | Report      : 00_Data_QC/02_trimmed_fastqc/multiqc_report.html\n",
      "\u001b[34m           multiqc\u001b[0m | MultiQC complete\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- Trimmed MultiQC complete. ---\n",
      "Check the 'multiqc_report.html' file in 00_Data_QC/02_trimmed_fastqc\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$trimmed_fastqc_dir\"\n",
    "# $1 = trimmed_fastqc_dir (variable from Cell 2)\n",
    "\n",
    "TRIMMED_QC_OUT=$1\n",
    "\n",
    "echo \"--- 5. Running MultiQC on TRIMMED FastQC Reports ---\"\n",
    "echo \"Target Directory: $TRIMMED_QC_OUT\"\n",
    "\n",
    "# Run MultiQC\n",
    "# -o $TRIMMED_QC_OUT : Output directory\n",
    "# $TRIMMED_QC_OUT : Directory to scan for reports\n",
    "multiqc -o $TRIMMED_QC_OUT $TRIMMED_QC_OUT\n",
    "\n",
    "echo \"--- Trimmed MultiQC complete. ---\"\n",
    "echo \"Check the 'multiqc_report.html' file in $TRIMMED_QC_OUT\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d1cf0acc-c56f-48e7-a126-9e9678c0b721",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
